fastjet/internal/heap/heap.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2017 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package heap
+
+// pair is an item in the heap. It contains the indices the two jets to be
+// clustered and their kt distance.
+type pair struct {
+	// jeti and jetj are the indices of the jets to be recombined.
+	jeti, jetj int
+	// dij is the kt distance for the two particles.
+	dij float64
+}
+
+// Heap contains a slice of items and the last index.
+type Heap struct {
+	n     int // n is the last index of the slice
+	items []pair
+}
+
+// NewHeap returns a heap pointer.
+func NewHeap() *Heap {
+	h := &Heap{n: 0}
+	// pair{} is a placeholder at h.items[0]
+	h.items = append(h.items, pair{})
```

Comment:
You could just as well write 'items: make([]pair, 1)' when creating 'h' and be done with it, right?